packages/wallet-sdk/src/util/provider.ts

```diff
@@ -43,6 +43,16 @@ export function getCoinbaseInjectedSigner(): Signer | undefined {
   return window.coinbaseWalletSigner;
 }
 
+function getTopFrameInjectedProvider(): CBInjectedProvider | undefined {
```

Comment:
how about calling it getInjectedProvider or something 
and encapsulate all window.ethereum logic?

Reply:
pushed, how does `getCipherWebViewInjectedProvider` sound?

Reply:
to be it seems to be just simple helper to retrieve injected provider from any of the window properties, either window or window.top
and cipher web view check happens the next line

Reply:
agree, `getInjectedEthereum`?